### PR TITLE
adding legacy opt in modal

### PR DESCRIPTION
**Pull request by jiahuang**
Connects #7335 

image

### CODE REVIEW

File: client/app/intake/components/LegacyOptInModal.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  radioOnChange = (value) => {
+    this.setState({
+      vacolsId: value
+    });
+  }
+
+  onAddIssue = () => {
+    // currently just adds the issue
```
Comment (jiahuang) [inline client/app/intake/components/LegacyOptInModal.jsx]: I think I also need to do a timelyness check here if "None of these match" is selected and show the timelyness alert module if it's not timely...?

File: spec/factories/legacy_appeal.rb [hunk 2]
```diff
@@ -2,10 +2,12 @@
   factory :legacy_appeal do
     transient do
       vacols_case nil
+      date nil
     end
 
     vacols_id { vacols_case.bfkey }
     vbms_id { vacols_case.bfcorlid }
+    nod_date { nil unless date }
```
Comment (jiahuang) [inline spec/factories/legacy_appeal.rb]: maybe this should be vacols_case.nod_date instead?

File: spec/feature/intake/supplemental_claim_spec.rb [hunk 3]
```diff
@@ -746,7 +746,21 @@ def check_row(label, text)
 
     context "with active legacy appeal" do
       before do
-        create(:legacy_appeal, vacols_case: create(:case, bfcorlid: "#{veteran.file_number}S"))
+        # create two legacy appeals with 2 issues each
```
Comment (jiahuang) [inline spec/feature/intake/supplemental_claim_spec.rb]: should this be pulled out into the intake helper?

File: client/app/intake/components/LegacyOptInModal.jsx (hunk 1 above)
Comment (leikkisa) [reply client/app/intake/components/LegacyOptInModal.jsx]: I think that timeliness is exempted if they choose "Yes" on opting into their legacy issues.

Comment (leikkisa) [reply client/app/intake/components/LegacyOptInModal.jsx]: I think that's correct if none of them match.

File: spec/feature/intake/supplemental_claim_spec.rb (hunk 3 above)
Comment (leikkisa) [reply spec/feature/intake/supplemental_claim_spec.rb]: I like that idea

File: app/models/decision_review.rb [hunk 4]
```diff
@@ -103,12 +103,13 @@ def mark_rating_request_issues_to_reassociate!
   end
 
   def serialized_legacy_issues
-    active_or_eligible_legacy_appeals.map do |legacy_appeal|
-      {
-        date: legacy_appeal.nod_date,
-        issues: legacy_appeal.issues.map(&:attributes)
-      }
-    end
+    active_or_eligible_legacy_appeals.reject { |legacy_appeal| legacy_appeal.issues.empty? }
```
Comment (shanear) [inline app/models/decision_review.rb]: i feel like an appeal with no issues should also not be eligible, and this logic should live a little deeper in the code?

File: spec/factories/legacy_appeal.rb (hunk 2 above)
Comment (shanear) [reply spec/factories/legacy_appeal.rb]: We shouldn't need to do this, we should just set `vacols_case.nod_date` like you suggested

**Pull request merged**